### ISSUE TITLE
no-top-level-await gets confused by anonymous function in same scope

### ISSUE THREAD

**Issue by edg2s** [ota-meshi/eslint-plugin-es-x#30]
The following triggers a false positive:
Config:
```json
{
	"root": true,
	"plugins": [ "es-x" ],
	"rules": {
		"es-x/no-top-level-await": "error"
	},
	"parserOptions": {
		"ecmaVersion": 2022
	},
	"env": {
		"es2022": true
	}
}
```

Code:
```js
async function foo () {
	dosomething( function () {}, 'foo' );

	await Promise.resolve();
}
```

If the line `dosomething` is removed, no error is triggered:
```js
async function foo () {
	await Promise.resolve();
}
```
